va/va.go

```diff
@@ -447,7 +449,8 @@ func (va *ValidationAuthorityImpl) validate(
 	}
 
 	// TODO(#1292): send into another goroutine
-	validationRecords, prob := va.validateChallenge(ctx, baseIdentifier, challenge)
+	ctxwithid := context.WithValue(ctx, accountIDKey{}, regid)
```

Comment:
You could alter the method signature for `va.validateChallenge` to take a 4th parameter of `regId int64` and remove smuggling this through the grpc context.